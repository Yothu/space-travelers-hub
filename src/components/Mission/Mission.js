import styled from 'styled-components';
import PropTypes from 'prop-types';
import '../Font.css';

const TableData = styled.td`
  padding: 8px;
  text-align: left;
  vertical-align: text-top;
  border: 1px solid #dddddd;
`;

const TableDataStatus = styled.td`
  padding: 8px;
  text-align: left;
  vertical-align: middle;
  border: 1px solid #dddddd;
`;

const JoinBtn = styled.button`
  color: black;
  padding: 0.3rem;
  margin: 0 0.6rem;
  border-radius: 3px;
  white-space: nowrap;
  border: 1px solid black;
  background-color: transparent;
`;

const LeaveBtn = styled.button`
  color: red;
  padding: 0.3rem;
  margin: 0 0.6rem;
  border-radius: 3px;
  white-space: nowrap;
  border: 1px solid red;
  background-color: transparent;
`;

const Active = styled.p`
  color: white;
  padding: 0.2rem;
  font-weight: bold;
  font-size: 0.6rem;
  border-radius: 5px;
  white-space: nowrap;
  background-color: #00a2b9;
`;

const NotActive = styled.p`
  color: white;
  padding: 0.2rem;
  font-weight: bold;
  font-size: 0.6rem;
  border-radius: 5px;
  white-space: nowrap;
  background-color: #6c737d;
`;

const descriptionStyle = {
  fontSize: '12px',
};

const nameStyle = {
  fontWeight: 'bold',
};

const Mission = ({ id, name, description }) => (
  <tr key={id}>
    <TableData style={nameStyle} className="font-1">{name}</TableData>
    <TableData style={descriptionStyle} className="font-1">{description}</TableData>
    <TableDataStatus>
      <NotActive className="font-1">NOT A MEMBER</NotActive>
      <Active className="font-1">Active Member</Active>
    </TableDataStatus>
    <TableDataStatus>
      <JoinBtn className="font-1">Join Mission</JoinBtn>
      <LeaveBtn className="font-1">Leave Mission</LeaveBtn>
    </TableDataStatus>
  </tr>
);

Mission.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default Mission;
