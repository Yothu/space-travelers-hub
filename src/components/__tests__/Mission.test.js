import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Mission from '../Mission/Mission';
import store from '../../redux/configureStore';

describe('Mission component', () => {
  it('it renders', () => {
    const mock = {
      id: 'abc',
      name: 'Start',
      reserved: false,
      description: 'this is the description',
    };

    const mission = renderer
      .create(
        <Provider store={store}>
          <Mission
            id={mock.id}
            key={mock.id}
            name={mock.name}
            reserved={mock.reserved}
            description={mock.description}
          />
        </Provider>,
      )
      .toJSON();
    expect(mission).toMatchSnapshot();
  });
});
