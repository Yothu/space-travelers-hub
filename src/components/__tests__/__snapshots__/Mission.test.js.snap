// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component it renders 1`] = `
<tr>
  <td
    className="sc-bdvvtL dIUcLi font-1"
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Start
  </td>
  <td
    className="sc-bdvvtL dIUcLi font-1"
    style={
      Object {
        "fontSize": "12px",
      }
    }
  >
    this is the description
  </td>
  <td
    className="sc-gsDKAQ jMXeGp"
  >
    <p
      className="sc-jRQBWg ejWCLK font-1"
    >
      NOT A MEMBER
    </p>
  </td>
  <td
    className="sc-gsDKAQ jMXeGp"
  >
    <button
      className="sc-dkPtRN kZuHOp font-1"
      onClick={[Function]}
    >
      Join Mission
    </button>
  </td>
</tr>
`;
